sklearn/feature_selection/_base.py

```diff
@@ -214,10 +223,16 @@ def _get_feature_importances(estimator, getter, transform_func=None, norm_order=
                 )
         else:
             getter = attrgetter(getter)
-    elif not callable(getter):
-        raise ValueError("`importance_getter` has to be a string or `callable`")
+        importances = getter(estimator)
 
-    importances = getter(estimator)
+    elif callable(getter):
+        if len(signature(getter).parameters) == 3:
+            importances = getter(estimator, X, y)
+        else:
+            importances = getter(estimator)
+
+    else:
+        raise ValueError("`importance_getter` has to be a string or `callable`")
```

Comment:
Hi @ClaudioSalvatoreArcidiacono , codecov is complaining that this ValueError is not checked in the tests.
Do you mind having a look? Thanks!

Reply:
Hi @cmarmo! thanks for your comment. I have removed the line not covered by tests, for more details check [my  comment below](https://github.com/scikit-learn/scikit-learn/pull/21935#discussion_r990813709).